src/components/Api.js

```diff
@@ -0,0 +1,103 @@
+export default class Api {
+  constructor(options) {
+    this.baseUrl = options.baseUrl;
+    this.headers = options.headers;
+  }
+
+  _checkServerResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error: ${res.status}`);
+    }
+  }
+
+  getInitialCards() {
+    return fetch(`${this.baseUrl}/cards`, {
+      headers: this.headers,
+    })
+      .then(this._checkServerResponse)
+      .catch((error) => {
+        console.error("Error fetching initial cards:", error);
+      });
+  }
+
+  getUserInfo() {
+    return fetch(`${this.baseUrl}/users/me`, {
+      headers: this.headers,
+    })
+      .then(this._checkServerResponse)
+      .catch((error) => {
+        console.error("Error fetching user info:", error);
+      });
+  }
+
+  updateProfile(name, about) {
+    return fetch(`${this.baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: {
+        ...this.headers,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name: name,
+        about: about,
+      }),
+    })
+      .then(this._checkServerResponse)
+      .catch((error) => {
```

Comment:
Needs correction:  Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```